src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Migrations/21.sql

```diff
@@ -0,0 +1,4104 @@
+    IF (@isDeleted = 1) BEGIN
+        SET @changeType = 2    /* DELETION */
+    END
+    ELSE BEGIN
+        IF (@version = 1) BEGIN
+            SET @changeType = 0 /* CREATION */
+        END
+        ELSE BEGIN
+            SET @changeType = 1 /* UPDATE */
+        END
+    END
+
+    INSERT INTO dbo.ResourceChangeData
+        (ResourceId, ResourceTypeId, ResourceVersion, ResourceChangeTypeId)
+    VALUES
+        (@resourceId, @resourceTypeId, @version, @changeType)
+END
+GO
+
+--
+-- STORED PROCEDURE
+--     FetchResourceChanges_2
+--
+-- DESCRIPTION
+--     Returns the number of resource change records from startId. The start id is inclusive.
+--
+-- PARAMETERS
+--     @startId
+--         * The start id of resource change records to fetch.
+--     @lastProcessedDateTime
+--         * The last checkpoint datetime.
+--     @pageSize
+--         * The page size for fetching resource change records.
+--
+-- RETURN VALUE
+--     Resource change data rows.
+--
+CREATE PROCEDURE dbo.FetchResourceChanges_2
+    @startId bigint,
+    @lastProcessedDateTime datetime2(7),
+    @pageSize smallint
+AS
+BEGIN
+
+    SET NOCOUNT ON;
+    
+    /* Given the fact that Read Committed Snapshot isolation level is enabled on the FHIR database, 
+       using the Repeatable Read isolation level table hint to avoid skipping resource changes 
+       due to interleaved transactions on the resource change data table.
+       In Repeatable Read, the select query execution will be blocked until other open transactions are completed
+       for rows that match the search condition of the select statement. 
+       A write transaction (update/delete) on the rows that match 
+       the search condition of the select statement will wait until the current transaction is completed. 
+       Other transactions can insert new rows that match the search conditions of statements issued by the current transaction.
+       But, other transactions will be blocked to insert new rows during the execution of the select query, 
+       and wait until the execution completes. */
+    SELECT TOP(@pageSize) Id,
+      Timestamp,
+      ResourceId,
+      ResourceTypeId,
+      ResourceVersion,
+      ResourceChangeTypeId
+      FROM dbo.ResourceChangeData WITH (REPEATABLEREAD)
+    WHERE Timestamp >= @lastProcessedDateTime AND Id >= @startId 
+    ORDER BY Timestamp ASC, Id ASC;
+END;
+GO
+
+/*************************************************************
+    Purge partition feature for resource change data
+**************************************************************/
+--
+-- STORED PROCEDURE
+--     ConfigurePartitionOnResourceChanges
+--
+-- DESCRIPTION
+--     Creates initial partitions for future datetimes on the resource change data table if they do not already exist.
+--
+-- PARAMETERS
+--     @numberOfFuturePartitionsToAdd
+--         * The number of partitions to add for future datetimes.
+--
+CREATE OR ALTER PROCEDURE dbo.ConfigurePartitionOnResourceChanges
```

Comment:
Since our codebase is backward compatible, the approach that we take to update the existing stored proc from previous version, is by adding incrementing version to it(for ex- [upsertResource ](https://github.com/microsoft/fhir-server/blob/main/src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Migrations/9.diff.sql#L49)Stored proc has been updated in few migrations) and then add logic in the code to use the relevant store proc based on the schema version.